apptax/taxonomie/commands/migrate_taxref/data/changes_detection/1.2_taxref_changes_detections_cas_actions.sql

```diff
@@ -1,172 +1,282 @@
+        WHERE cas = 'split'
+            AND i_cd_ref != f_cd_ref
+    ) AS l
+    LEFT JOIN (
+        SELECT *
+        FROM tmp_taxref_changes.comp_grap
+        WHERE cas = 'split'
+            AND i_cd_ref = f_cd_ref
+    ) AS k
+    ON k.i_cd_ref = l.i_cd_ref
+) AS a
+WHERE a.i_cd_ref = c.i_cd_ref
+    AND a.f_cd_ref = c.f_cd_ref ;
+
+
+-- ----------------------------------------------------------------------
+-- Medium & attributs: case "merge", detect "no change" for attributs
+WITH cdref_maintained AS (
+    SELECT
+        i_cd_ref AS cd_ref,
+        i_array_agg,
+        f_array_agg
+    FROM tmp_taxref_changes.comp_grap
+    WHERE i_cd_ref = f_cd_ref
+),
+no_cdref_change AS (
+    SELECT
+        cg.i_cd_ref,
+        cg.f_cd_ref
+    FROM tmp_taxref_changes.comp_grap AS cg
+        JOIN cdref_maintained AS cm_i
+            ON (cm_i.cd_ref = cg.i_cd_ref AND cg.i_array_agg = cm_i.i_array_agg)
+    WHERE cg.cas = 'merge'
+        AND cg.grappe_change ILIKE '%cas3: f_cd_ref%'
+           AND cg.i_cd_ref != cg.f_cd_ref
+           AND cg.att_nb > 0
 )
-UPDATE tmp_taxref_changes.comp_grap c SET action = 'Conflicts with attributes : ' || atts
-FROM conflict_atts_text a
-WHERE a.f_cd_ref = c.f_cd_ref AND  cas = 'merge';
-
-UPDATE tmp_taxref_changes.comp_grap c SET action = 'Merge attributes if exists'
-WHERE cas = 'merge' AND action IS NULL;
-
--- Split and merge
+UPDATE tmp_taxref_changes.comp_grap AS c SET
+    "action" = 'Do not change attributes and media'
+FROM no_cdref_change AS ncc
+WHERE c.i_cd_ref = ncc.i_cd_ref
+    AND c.f_cd_ref = ncc.f_cd_ref ;
 
-UPDATE tmp_taxref_changes.comp_grap c SET action = 'Keep attributes and medium'
-WHERE  cas = 'split and merge' AND i_cd_ref = f_cd_ref;
 
-UPDATE tmp_taxref_changes.comp_grap c SET action = 'Loose attributes and medium now attach to ' || COALESCE(cd_ref_attr::varchar, 'No one')
+-- ----------------------------------------------------------------------
+-- Medium & attributs: case "merge", detect conflicts for attributs
+WITH atts AS (
+    SELECT DISTINCT *
+    FROM taxonomie.cor_taxon_attribut AS a
+        JOIN tmp_taxref_changes.comp_grap AS c
+            ON a.cd_ref = c.i_cd_ref
+    WHERE valeur_attribut != '{}'
+        AND valeur_attribut != ''
+        AND cas = 'merge'
+        AND i_cd_ref != f_cd_ref
+),
+conflict_atts AS (
+    SELECT
+        f_cd_ref,
+        id_attribut,
+        count(DISTINCT valeur_attribut)
+    FROM atts
+    GROUP BY f_cd_ref, id_attribut
+    HAVING count(DISTINCT valeur_attribut) > 1
+),
+conflict_atts_text AS (
+    SELECT
+        f_cd_ref,
+        string_agg(nom_attribut::varchar, ', ') AS atts
+    FROM conflict_atts AS c
+        JOIN taxonomie.bib_attributs AS a
+            ON a.id_attribut = c.id_attribut
+    GROUP BY f_cd_ref
+)
+UPDATE tmp_taxref_changes.comp_grap AS c SET
+    "action" = 'Conflicts with attributes : ' || atts
+FROM conflict_atts_text AS a
+WHERE a.f_cd_ref = c.f_cd_ref
+    AND cas = 'merge' ;
```

Comment:
Est ce qu'il ne faudrait pas ajouter `AND "action" IS NULL` ici ?

Reply:
Finalement, après relecture du code SQL, je dirai que non car le champ "action" est potentielle déjà rempli. Par exemple avec 'Update cd_ref no changes for attributes and medium'.